main.tf

```diff
@@ -155,6 +155,12 @@ resource "google_alloydb_instance" "primary" {
     cpu_count = var.primary_instance.machine_cpu_count
   }
 
+  client_connection_config {
```

Comment:
change following block to dynamic block

```
dynamic "client_connection_config" {
  for_each = coalesce(var.primary_instance.ssl_mode, var.primary_instance.require_connectors) == null ? [] : ["client_connection_config"]
    content {
      ssl_config {
        ssl_mode            = coalesce(var.primary_instance.ssl_mode)
        require_connectors  = coalesce(var.primary_instance.require_connectors)

      }
    }
}
```

Reply:
Fixed but required_connectors order is bit different than suggested.